Sources/Extensions/NSDictionary+Toggles.swift

```diff
@@ -0,0 +1,39 @@
+//  NSDictionary+Toggles.swift
+
+import Foundation
+
+extension NSDictionary {
+    public enum ToggleConversionError: Error {
```

Comment:
With the above change, I think this will have to be moved out of the extension in order to be referenced from the tests. It can now also be internal and have a single case with an associated value.

```
enum ToggleConversionError: Error, Equatable {
    case unsupportedValueType(String)
}
```